tests/components/minecraft_server/test_sensor.py

```diff
@@ -0,0 +1,362 @@
+            "sensor.minecraft_server_latency",
+        ),
+        (
+            "java_mock_config_entry",
+            JavaServer,
+            TEST_JAVA_STATUS_RESPONSE,
+            "sensor.minecraft_server_players_online",
+        ),
+        (
+            "java_mock_config_entry",
+            JavaServer,
+            TEST_JAVA_STATUS_RESPONSE,
+            "sensor.minecraft_server_world_message",
+        ),
+        (
+            "java_mock_config_entry",
+            JavaServer,
+            TEST_JAVA_STATUS_RESPONSE,
+            "sensor.minecraft_server_version",
+        ),
+        (
+            "bedrock_mock_config_entry",
+            BedrockServer,
+            TEST_BEDROCK_STATUS_RESPONSE,
+            "sensor.minecraft_server_latency",
+        ),
+        (
+            "bedrock_mock_config_entry",
+            BedrockServer,
+            TEST_BEDROCK_STATUS_RESPONSE,
+            "sensor.minecraft_server_players_online",
+        ),
+        (
+            "bedrock_mock_config_entry",
+            BedrockServer,
+            TEST_BEDROCK_STATUS_RESPONSE,
+            "sensor.minecraft_server_world_message",
+        ),
+        (
+            "bedrock_mock_config_entry",
+            BedrockServer,
+            TEST_BEDROCK_STATUS_RESPONSE,
+            "sensor.minecraft_server_version",
+        ),
+        (
+            "bedrock_mock_config_entry",
+            BedrockServer,
+            TEST_BEDROCK_STATUS_RESPONSE,
+            "sensor.minecraft_server_map_name",
+        ),
+        (
+            "bedrock_mock_config_entry",
+            BedrockServer,
+            TEST_BEDROCK_STATUS_RESPONSE,
+            "sensor.minecraft_server_game_mode",
+        ),
+    ],
+)
+async def test_sensor_update_failure(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    server: JavaServer | BedrockServer,
+    status_response: JavaStatusResponse | BedrockStatusResponse,
+    entity_id: er.EntityRegistry,
+    request: pytest.FixtureRequest,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test failed sensor update."""
+    mock_config_entry = request.getfixturevalue(mock_config_entry)
+    mock_config_entry.add_to_hass(hass)
+
+    with patch(
+        f"homeassistant.components.minecraft_server.api.{server.__name__}.lookup",
+        return_value=server(host=TEST_HOST, port=TEST_PORT),
+    ), patch(
+        f"homeassistant.components.minecraft_server.api.{server.__name__}.async_status",
+        return_value=status_response,
+    ):
+        assert await hass.config_entries.async_setup(mock_config_entry.entry_id)
+        await hass.async_block_till_done()
+
+    future = dt_util.utcnow() + timedelta(minutes=2)
+    with patch(
+        f"homeassistant.components.minecraft_server.api.{server.__name__}.async_status",
+        side_effect=OSError,
+    ):
+        async_fire_time_changed(hass, future)
+        await hass.async_block_till_done()
+        assert hass.states.get(entity_id) == snapshot
```

Comment:
Snapshot tests are nice, but I think it's a bit too much in this PR. I think this check could just be

```suggestion
        assert hass.states.get(entity_id).state == STATE_UNAVAILABLE
```

Reply:
It's very tempting to use snapshots for everything as it's so easy to use 😄 👍

Reply:
A guideline I tend to use is, if I need to compare a big data structure, use snapshots. If I want to test one aspect of a data structure, just test that aspect.